solang-parser/src/pt.rs

```diff
@@ -597,7 +597,8 @@ pub enum Statement {
     },
     Assembly {
         loc: Loc,
-        assembly: AssemblyBlock,
+        assembly: Vec<AssemblyStatement>,
+        dialect: Option<String>,
```

Comment:
Very much a nitpick, but in the parse tree I would place fields in the order they appear in the original source, so `dialect` first. Also I think `statements` would be a better name than `assembly`.